From d3027a227bc0f603a5b650d01f97ee1dec515be5 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 18 Sep 2015 00:28:45 -0400
Subject: [PATCH] snmplib/keytools.c: Don't check for return from
 EVP_MD_CTX_init()

EVP_MD_CTX_init() API returns void, it fixes errors with new compilers

snmplib/keytools.c: In function 'generate_Ku': error: invalid use of void expression

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
Upstream-Status: Submitted [https://sourceforge.net/p/net-snmp/patches/1317/]
 snmplib/keytools.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/snmplib/keytools.c b/snmplib/keytools.c
index 0ccb3a6..880fc14 100644
--- a/snmplib/keytools.c
+++ b/snmplib/keytools.c
@@ -153,8 +153,7 @@ generate_Ku(const oid * hashtype, u_int hashtype_len,
     ctx = EVP_MD_CTX_create();
 #else
     ctx = malloc(sizeof(*ctx));
-    if (!EVP_MD_CTX_init(ctx))
-        return SNMPERR_GENERR;
+    EVP_MD_CTX_init(ctx);
 #endif
 #ifndef NETSNMP_DISABLE_MD5
     if (ISTRANSFORM(hashtype, HMACMD5Auth)) {
-- 
2.5.2

